### PR TITLE
Update service/client request/response API error returns

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Comment by Lobotuerk** [comment]
CI up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12454/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7421/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10165/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12373/)

File: rmw_connext_cpp/src/rmw_request.cpp [hunk 1]
```diff
@@ -28,19 +28,16 @@ rmw_send_request(
...
-  if (!ros_request) {
-    RMW_SET_ERROR_MSG("ros request handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(ros_request, RMW_RET_INVALID_ARGUMENT);
+
+  RMW_CHECK_ARGUMENT_FOR_NULL(sequence_id, RMW_RET_INVALID_ARGUMENT);
 
   ConnextStaticClientInfo * client_info = static_cast<ConnextStaticClientInfo *>(client->data);
   if (!client_info) {
```
Comment (hidmic) [inline rmw_connext_cpp/src/rmw_request.cpp:43]: @Lobotuerk We can reduce LOC count a bit here. This check and the two following ones make no sense IMHO, as those imply that:

- the caller is not using the API outside the spec, or
- the implementation is logically incorrect, or
- the process memory space is corrupt.

File: rmw_connext_cpp/src/rmw_request.cpp [hunk 2]
```diff
@@ -28,19 +28,16 @@ rmw_send_request(
   const void * ros_request,
   int64_t * sequence_id)
 {
-  if (!client) {
-    RMW_SET_ERROR_MSG("client handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
+
```
Comment (hidmic) [inline rmw_connext_cpp/src/rmw_request.cpp:32]: @Lobotuerk nit: unnecessary blank line.

File: rmw_connext_cpp/src/rmw_request.cpp [hunk 3]
```diff
@@ -28,19 +28,16 @@ rmw_send_request(
...
     client->implementation_identifier, rti_connext_identifier,
-    return RMW_RET_ERROR)
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
 
-  if (!ros_request) {
-    RMW_SET_ERROR_MSG("ros request handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(ros_request, RMW_RET_INVALID_ARGUMENT);
+
```
Comment (hidmic) [inline rmw_connext_cpp/src/rmw_request.cpp:39]: @Lobotuerk nit:unnecessary blank line.

File: rmw_connext_cpp/src/rmw_request.cpp [hunk 4]
```diff
@@ -28,19 +28,16 @@ rmw_send_request(
...
-    RMW_SET_ERROR_MSG("client handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
+
   RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
     client handle,
     client->implementation_identifier, rti_connext_identifier,
-    return RMW_RET_ERROR)
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
```
Comment (hidmic) [inline rmw_connext_cpp/src/rmw_request.cpp:37]: @Lobotuerk nit:unnecessary blank line.

Comment (Lobotuerk) [reply rmw_connext_cpp/src/rmw_request.cpp]: Updated

File: rmw_connext_cpp/src/rmw_request.cpp (hunk 3 above)
Comment (Lobotuerk) [reply rmw_connext_cpp/src/rmw_request.cpp]: Updated

File: rmw_connext_cpp/src/rmw_request.cpp (hunk 2 above)
Comment (Lobotuerk) [reply rmw_connext_cpp/src/rmw_request.cpp]: Updated

File: rmw_connext_cpp/src/rmw_request.cpp (hunk 1 above)
Comment (Lobotuerk) [reply rmw_connext_cpp/src/rmw_request.cpp:40]: Should this be added into another PR, as I don't think completely relates to this "Error Update" PR

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

**Comment by Lobotuerk** [comment]
CI up to `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12501/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7469/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10213/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12421/)

File: rmw_connext_cpp/src/rmw_request.cpp [hunk 5]
```diff
@@ -69,27 +63,14 @@ rmw_take_request(
   void * ros_request,
   bool * taken)
 {
-  if (!service) {
-    RMW_SET_ERROR_MSG("service handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(service, RMW_RET_INVALID_ARGUMENT);
   RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
-    service handle,
```
Comment (Blast545) [inline rmw_connext_cpp/src/rmw_request.cpp:77]: Why is this being removed?

Comment (Blast545) [reply rmw_connext_cpp/src/rmw_request.cpp:77]: I can see in the client part (Line 33) the `handle` is left there

Comment (Blast545) [reply rmw_connext_cpp/src/rmw_request.cpp:77]: Based on the rest of the code I think you may want to remove it there.

**Review by Blast545 — APPROVED** [review]
Check comment about handle, LGTM with green CI (aarch is red atm)

**Comment by Lobotuerk** [comment]
CI job after changes, up to `rcl` and `test_rmw_implementations`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12508/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7476/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10220/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12428/)

**Comment by Lobotuerk** [comment]
All CI green, merging

**Pull request merged**